angular-autocomplete.js

```diff
@@ -18,27 +18,37 @@
     return angular.module('mp.autocomplete', []).directive('autocomplete', [
         '$window', '$q', '$compile', '$parse', '$timeout',
         function($window, $q, $compile, $parse, $timeout) {
+            // Introduce custom elements for IE8
+            $window.document.createElement('autocomplete');
+
             var TEMPLATE = '' +
-                '<div class="angular-autocomplete" ng-if="isOpen()">' +
+                '<div class="angular-autocomplete" ng-show="isOpen()" tabindex="0">' +
                 '  <ul class="_suggestions">' +
                 '    <li ng-repeat="match in matches" ' +
                 '        class="_suggestion"' +
                 '        ng-class="{ \'-selected\': selectedIndex == $index }" ' +
                 // use mousedown to stay clear of input losing focus when suggestion is clicked
-                '        ng-mousedown="select($index)">{{ match }}</li>' +
+                '        ng-mousedown="select($index)" ' +
+                '        ng-mousemove="setSelectedIndex($index)">{{ match }}</li>' +
                 '  </ul>' +
                 '</div>';
 
             var HOT_KEYS = [9, 13, 27, 38, 40];
 
             return {
-                restrict: 'A',
+                restrict: 'E',
+                template: TEMPLATE,
+                replace: true,
                 require: 'ngModel',
                 scope: {
-                    onSuggestionSelected: '&',
-                    autocomplete: '='
+                    onSelectionComplete: '=',
```

Comment:
This seems like the easiest approach to pass functions to the directive. Using `@` will require interpolation at some point, and `&` is just a reference to the parent method. Perhaps I will change `onSelectionComplete` back to using `&`.